lib/HttpClient.js

```diff
@@ -105,6 +107,16 @@ function RequestTimeoutError(ms) {
 }
 util.inherits(RequestTimeoutError, Error);
 
+function TooManyRedirectsError(redirectNumber) {
+    if (Error.captureStackTrace) {
+        Error.captureStackTrace(this, TooManyRedirectsError);
+    }
+
+    this.message = 'Aborted after ' + redirectNumber + ' redirects';
+    this.name = 'TooManyRedirectsError';
+}
+util.inherits(TooManyRedirectsError, Error);
```

Comment:
@DonutEspresso should these be using the `makeConstructor` from Restify Errors? Or just stick with the established pattern?